src/model/test/CoilWaterHeatingDesuperheater_GTest.cpp

```diff
@@ -67,6 +67,13 @@ TEST_F(ModelFixture, CoilWaterHeatingDesuperheater_DefaultConstructor) {
       exit(0);
     },
     ::testing::ExitedWithCode(0), "");
+
+  // Passing an incompatible schedule: we expect the Ctor to throw, and no object to be created
+  Model model;
+  EXPECT_EQ(0, model.getConcreteModelObjects<CoilWaterHeatingDesuperheater>().size());
+  auto alwaysOn = model.alwaysOnDiscreteSchedule();
+  EXPECT_ANY_THROW(CoilWaterHeatingDesuperheater(model, alwaysOn));
+  EXPECT_EQ(0, model.getConcreteModelObjects<CoilWaterHeatingDesuperheater>().size());
```

Comment:
Test for the Ctor issue. It fails before: OS_ASSERT messages are issued but there are 1 object created anyways.